docs/src/pages/components/data-grid/editing/CellEditServerSidePersistence.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  GridCellEditCommitParams,
+  GridColumns,
+  GridRowId,
+  GridRowsProp,
+  useGridApiRef,
+} from '@mui/x-data-grid-pro';
+import {
+  randomCreatedDate,
+  randomTraderName,
+  randomUpdatedDate,
+} from '@mui/x-data-grid-generator';
+import Snackbar from '@mui/material/Snackbar';
+import Alert, { AlertProps } from '@mui/material/Alert';
+
+interface User {
+  name: string;
+  age: number;
+  id: GridRowId;
+  dateCreated: Date;
+  lastLogin: Date;
+}
+
+const useFakeMutation = () => {
+  return React.useCallback(
+    (user: Partial<User>) =>
+      new Promise((resolve) =>
+        setTimeout(() => {
+          resolve(user);
+        }, 200),
+      ),
+    [],
+  );
+};
+
+export default function CellEditServiceSidePersistence() {
+  const mutateRow = useFakeMutation();
+  const apiRef = useGridApiRef();
+  const [snackbar, setSnackbar] = React.useState<Pick<
+    AlertProps,
+    'children' | 'severity'
+  > | null>(null);
+
+  const handleCloseSnackbar = () => setSnackbar(null);
+
+  const handleCellEditCommit = React.useCallback(
+    async (params: GridCellEditCommitParams) => {
+      // Get the row old value before committing
+      const oldRow = apiRef.current.getRow(params.id)!;
```

Comment:
You can get the row from the `rows` prop and make this demo Community-compatible.

Reply:
Done